### DISCUSSION TITLE
Event for modifying the options used for an option field's input

### DISCUSSION THREAD

**Discussion by friartuck6000** [craftcms/cms#12212] [Ideas]
The Matrix field exposes a really helpful event ([`EVENT_SET_FIELD_BLOCK_TYPES`](https://github.com/craftcms/cms/blob/main/src/fields/Matrix.php#L692)), which makes it possible to change which block types are available in the editor for a given element.

We have a few use cases where it would be really valuable if we could do the same with option fields like Dropdown and Checkboxes—change which options are actually available in the editor in a given context.

I have a PR for this ready to submit if y'all are open to reviewing it.

**Comment by brandonkelly** [comment]
Can’t promise we’ll accept it but go ahead and submit the PR! We’ll consider it from there.
